src/basic/proc-cmdline.c

```diff
@@ -119,7 +120,9 @@ int proc_cmdline_parse(proc_cmdline_parse_t parse_item, void *data, ProcCmdlineF
 
         /* We parse the EFI variable first, because later settings have higher priority. */
 
-        if (!FLAGS_SET(flags, PROC_CMDLINE_IGNORE_EFI_OPTIONS)) {
+        if (!FLAGS_SET(flags, PROC_CMDLINE_IGNORE_EFI_OPTIONS) &&
+            (!FLAGS_SET(flags, PROC_CMDLINE_IGNORE_EFI_OPTIONS_BY_ENV) ||
+             getenv_bool("SYSTEMD_PARSE_EFI") != 0)) {
```

Comment:
hmm, why do we have to opt into this? it appears to me that when the env var is set it should be honoured, without this requiring a special flag to be passed.

i.e. i am fine with where this patch goes, but i'd simplify it by getting rid of the PROC_CMDLINE_IGNORE_EFI_OPTIONS_BY_ENV flag and making it the only behaviour supported.

(also, sounds like something where to getenv_bool_secure() would have been the better choice)